File: sorting and searching/insertionsort.py
def insertionsort(arr,n):
    for i in range(n):
        for j in range(i-1,-1,-1):
            if(arr[j]>arr[j+1]):
                arr[j],arr[j+1]=arr[j+1],arr[j]
            else:
                break 
n=int(input("Enter the size of the array:"))
arr=[]
print("Enter elements of the array:")
for i in range(n):
    l=int(input())
    arr.append(l)
insertionsort(arr,n)
print("Sorted array in Ascending order by using Insertion Sort:")
print(arr)

File: sorting and searching/insertionsort.cpp
#include<iostream>
using namespace std;
void insertionsort(int arr[],int n){
    for(int i=1;i<n;i++){
        for(int j=i-1;j>=0;j--){
            if(arr[j]>arr[j+1]){
                int temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
            else
            {
                break;
            }
            
        }
    }

} 
int main(){
    int i,n;
    cout<<"Enter the size of the array:";
    cin>>n;
    cout<<"Enter the elements of the array:";
    int arr[n];
    for(i=0;i<n;i++){
        cin>>arr[i];
    }
    insertionsort(arr,n);
    cout<<"Sorted array in Ascending order by using Insertion Sort:"<<endl;
    for(i=0;i<n;i++)
        cout<<arr[i]<<" ";
    cout<<"\n";
}
